src/Workspaces/Core/Desktop/Workspace/Host/TemporaryStorage/TemporaryStorageServiceFactory.MemoryMappedInfo.cs

```diff
@@ -115,85 +106,91 @@ public Stream CreateWritableStream()
                 }
             }
 
-            private void StreamDisposed()
+            public void Dispose()
             {
-                lock (_memoryMappedFile)
+                Dispose(true);
+                GC.SuppressFinalize(this);
+            }
+
+            private void Dispose(bool disposing)
+            {
+                if (disposing)
                 {
-                    _streamCount--;
-                    if (_streamCount == 0 && _accessor != null)
+                    lock (_memoryMappedFile)
                     {
-                        _accessor.Dispose();
-                        _accessor = null;
+                        if (_accessor != null)
+                        {
+                            // dispose accessor it owns.
```

Comment:
with what you changed, stream given out will always work regardless accessor, memory map file being already disposed right? so this comment seems no longer needed.